src/pokemonClass.ts

```diff
@@ -0,0 +1,69 @@
+export type pokemonType =
+  | "grass"
+  | "poison"
+  | "fire"
+  | "water"
+  | "bug"
+  | "normal"
+  | "flying"
+  | "electric"
+  | "ground"
+  | "fairy"
+  | "psychic"
+  | "fighting"
+  | "rock";
+
+export abstract class Pokemon {
+  constructor() {}
+  abstract readonly _name: string;
+  abstract readonly _image: string;
+  abstract readonly _maxHP: number;
+  abstract _currentHP: number;
+  abstract _attackPoints: number;
+  abstract _defensePoints: number;
+  abstract readonly _types: pokemonType[];
+  abstract readonly _moves: { moveName: string; moveType: pokemonType }[];
+  abstract _isAlive: boolean = true;
+
+  get name() {
+    return this._name;
+  }
+
+  get image() {
+    return this._image;
+  }
+
+  get currentHP() {
+    return this._currentHP;
+  }
+
+  set currentHP(newHP) {
+    if (newHP <= 0) {
+      this._currentHP = 0;
+    }
+    if (newHP >= this._maxHP) {
+      this._currentHP = this._maxHP;
+    }
+    this._currentHP = newHP;
+  }
```

Comment:
I see that you probably think, that objects cannot have different methods/functions than get and set. It is not true :)
You can define function that would be a good replacement for this one. 
Instead of deciding what value "new hp" would be, you can (and should) create different methods, like: "subtract hp" and "heal". Setting specific value, from the point of the game is not needed. All you need to do is subtract from current hp (and check if this would not be less than 0) and heal pokemon (which will set pokemon hp again to max)